clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ConsumerPollTrackerReportingTask.java

```diff
@@ -0,0 +1,63 @@
+package com.linkedin.davinci.kafka.consumer;
+
+import com.linkedin.venice.pubsub.api.PubSubTopicPartition;
+import com.linkedin.venice.utils.Time;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.Logger;
+
+
+public class ConsumerPollTrackerReportingTask implements Runnable {
+  private static final long DEFAULT_CONSUMER_POLL_TRACKER_CHECK_CYCLE = TimeUnit.MINUTES.toMillis(1);
+  private static final String WARNING_MESSAGE_PREFIX = "Consumer poll tracker found stale topic partitions: ";
+  private final ConsumerPollTracker consumerPollTracker;
+  private final Time time;
+  private final long staleThresholdMs;
+  private final Logger LOGGER;
+  private final StringBuilder stringBuilder = new StringBuilder();
+  private volatile boolean reportingTaskStopped = false;
+
+  public ConsumerPollTrackerReportingTask(
+      ConsumerPollTracker consumerPollTracker,
+      Time time,
+      long staleThresholdMs,
+      Logger logger) {
+    this.consumerPollTracker = consumerPollTracker;
+    this.time = time;
+    this.staleThresholdMs = staleThresholdMs;
+    this.LOGGER = logger;
+  }
+
+  public void stop() {
+    reportingTaskStopped = true;
+  }
+
+  @Override
+  public void run() {
```

Comment:
It seems this task extended the functionality of previous stuck consumer stats and fix: `getStuckConsumerDetectionAndRepairRunnable`. Maybe we could add the logic into that thread to avoid creating too many threads to increase complexity of code. Besides, if we have metric to indicate this per-partition issue happening per server host, it would also be helpful to tracking.